pkg/kapp/app/recorded_app.go

```diff
@@ -152,6 +152,32 @@ func (a *RecordedApp) CreateOrUpdate(labels map[string]string, isDiffRun bool) (
 		return false, a.updateApp(app, labels)
 	}
 
+	if prevAppName == "" {
+		return true, a.create(labels, isDiffRun)
+	}
+
+	app, foundMigratedPrevApp, err := a.find(prevAppName + AppSuffix)
+	if err != nil {
+		return false, err
+	}
+
+	if foundMigratedPrevApp {
+		a.isMigrated = true
+		return false, a.renameConfigMap(app, a.fqName(), a.nsName)
+	}
+
+	app, foundNonMigratedPrevApp, err := a.find(prevAppName)
+	if err != nil {
+		return false, err
+	}
+
```

Comment:
nit: I think we can remove newline here as all this is for one scenario. I think same can be done for others as well.

Reply:
Done